test/lib/tmpdir.cc

```diff
@@ -31,9 +31,27 @@ tmpdir::sweeper::~sweeper() {
     }
 }
 
-tmpdir::tmpdir()
-    : _path(fs::temp_directory_path() / fs::path(fmt::format(FMT_STRING("scylla-{}"), utils::make_random_uuid()))) {
-    fs::create_directories(_path);
+tmpdir::tmpdir() {
+    auto tmp = fs::temp_directory_path();
+    for (;;) {
+        // Reduce the path length of the created tmp dir. This might seem
+        // silly when running with base TMPDIR=/tmp or similar, but
+        // in a lot of CI testing, TMPDIR will be a loooooong path into
+        // jenkins workdirs or similar -> this path will be 100+ chars long.
+        // Again, this should most often not be a problem, _but_ if we
+        // for example run something like a sub process of a python server,
+        // which will try to create various unix sockets et al for its
+        // operations, the TMPDIR base for this must not exceed 107 chars.
+        // Note: converting UUID to string first, because for some reason
+        // our UUID formatter does not respect width/precision. Feel free to
+        // change once it does.
+        _path = tmp / fmt::format("scylla-{:.8}", fmt::to_string(utils::make_random_uuid()));
+        // Note: this is a slight improvement also, in that we ensure the dir 
+        // we use is actually created by us.
+        if (fs::create_directories(_path)) {
+            break;
```

Comment:
Since it is a random path - maybe log its name to ease debugging before breaking?

Reply:
We never did in neither enterprise nor master code.

Reply:
(This is not a change, just merging a commit only existing in enterprise)